server/controllers/subject.js

```diff
@@ -0,0 +1,183 @@
+/* eslint-disable sort-keys */
+/* eslint-disable consistent-return */
+const HttpStatus = require('http-status-codes');
+const Subject = require("../models/Subject");
+
+const { BAD_REQUEST, OK, INTERNAL_SERVER_ERROR, CREATED, NOT_FOUND } = HttpStatus;
+
+// Create and Save a new Subject
+exports.create = async(req, res) => {
+    try {
+
+        const { name, description } = req.body;
+        // Validate request
+        if (!name) {
+            return res.status(BAD_REQUEST)
+                .send({
+                    status: "fail",
+                    data: { "name": "Subject name is required" },
+                });
+        }
+
+        // Create a Subject
+        const subject = new Subject({
+            description,
+            name: name || "Undefined Subject",
```

Comment:
you would never have an undefined `name` here so there is no need for `|| "Undefined Subject"` check. Just simply write:
```
{
    description,
    name,
}
```